eiger_io/fs_handler.py

```diff
@@ -1,69 +1,172 @@
+
+    @property
+    def frame_shape(self):
+        return self[0].shape
 
+    @property
+    def pixel_type(self):
+        return self[0].dtype
 
-EIGER_MD_DICT = {
-    'y_pixel_size': 'entry/instrument/detector/y_pixel_size',
-    'x_pixel_size': 'entry/instrument/detector/x_pixel_size',
-    'detector_distance': 'entry/instrument/detector/detector_distance',
-    'incident_wavelength': 'entry/instrument/beam/incident_wavelength',
-    'frame_time': 'entry/instrument/detector/frame_time',
-    'beam_center_x': 'entry/instrument/detector/beam_center_x',
-    'beam_center_y': 'entry/instrument/detector/beam_center_y',
-    'count_time': 'entry/instrument/detector/count_time',
-    'pixel_mask': 'entry/instrument/detector/detectorSpecific/pixel_mask',
-}
+    @property
+    def dtype(self):
+        return self.pixel_type
 
+    @property
+    def shape(self):
+        return self.frame_shape
 
-class LazyEigerHandler(HandlerBase):
-    specs = {'AD_EIGER'} | HandlerBase.specs
-    def __init__(self, fpath, frame_per_point, mapping=None):
+    def close(self):
+        self._handle.close()
+
+
+class EigerHandler(HandlerBase):
+    EIGER_MD_LAYOUT = {
+        'y_pixel_size': 'entry/instrument/detector/y_pixel_size',
+        'x_pixel_size': 'entry/instrument/detector/x_pixel_size',
+        'detector_distance': 'entry/instrument/detector/detector_distance',
+        'incident_wavelength': 'entry/instrument/beam/incident_wavelength',
+        'frame_time': 'entry/instrument/detector/frame_time',
+        'beam_center_x': 'entry/instrument/detector/beam_center_x',
+        'beam_center_y': 'entry/instrument/detector/beam_center_y',
+        'count_time': 'entry/instrument/detector/count_time',
+        'pixel_mask': 'entry/instrument/detector/detectorSpecific/pixel_mask',
+    }
+    specs = {'AD_EIGER2', 'AD_EIGER'}
+
+    def __init__(self, fpath, images_per_file=None, frame_per_point=None):
+        ''' Initializer for Eiger handler.
+
+            Parameters
+            ----------
+            fpath : str
+                the partial file path
+
+            images_per_file : int, optional
+                images per file. If not set, must set frame_per_point
+
+            frame_per_point : int, optional. If not set, must set
+                images_per_file
+
+            This one is backwards compatible for both versions of resources
+            saved in databroker. Old resources used 'frame_per_point' as a
+            kwarg. Newer resources call this 'images_per_file'.
+        '''
+        print("filepath : {}".format(fpath))
         # create pims handler
-        self.vals_dict = EIGER_MD_DICT.copy()
-        if mapping is not None:
-            self.vals_dict.update(mapping)
         self._base_path = fpath
-        self.fpp = frame_per_point
+        if images_per_file is None and frame_per_point is None:
+            errormsg = "images_per_file and frame_per_point both set"
```

Comment:
Stylistic note -- we usually go with:

```python
("blah blah blah "
 "blah blah blah "
 "blah blah blah ")
```

or multiple strings with triple-quotes.